manage.py

```diff
@@ -1,29 +1,107 @@
+import os
 import sys
+import click
 import logging
 
-from listenbrainz_spark.stats import user
-from listenbrainz_spark.recommendations import create_dataframes
-from listenbrainz_spark.recommendations import train_models
-from listenbrainz_spark.recommendations import recommend
-from listenbrainz_spark.recommendations import candidate_sets
+import listenbrainz_spark
+from listenbrainz_spark import path
+from listenbrainz_spark import utils
+from listenbrainz_spark import config
+from listenbrainz_spark import hdfs_connection
 
-if __name__ == '__main__':
-    if len(sys.argv) != 2:
-        print("Usage: manage.py <module_name>")
+from hdfs.util import HdfsError
+from py4j.protocol import Py4JJavaError
+
+@click.group()
+def cli():
+    pass
+
+@cli.command(name='init_dir')
+@click.option('--rm', is_flag=True, help='Delete existing directories from HDFS.')
+@click.option('--f', is_flag=True, help='Delete existing directories from HDFS recursively.')
+@click.option('--create_dir', is_flag=True, help='Create directories in HDFS.')
+def init_dir(rm, f, create_dir):
+    """ Create directories in HDFS to run the recommendation engine.
+    """
+    try:
+        listenbrainz_spark.init_spark_session('Manage Directories')
+    except Py4JJavaError as err:
+        logging.error('{}\n{}\nAborting...'.format(str(err), err.java_exception))
         sys.exit(-1)
 
+    hdfs_connection.init_hdfs(config.HDFS_HTTP_URI)
+    if rm:
+        try:
+            utils.delete_dir(path.RECOMMENDATION_PARENT_DIR)
+            utils.delete_dir(path.CHECKPOINT_DIR)
+            logging.info('Successfully deleted directories.')
+        except HdfsError as err:
+            logging.error('{}: Some/all directories are non-empty. Try "--f" to delete recursively.'.format(
+                type(err).__name__))
+            logging.warning('Deleting directory recursively will delete all the recommendation data.')
+            sys.exit(-1)
+
+    if f:
+        utils.delete_dir(path.RECOMMENDATION_PARENT_DIR, recursive=True)
+        utils.delete_dir(path.CHECKPOINT_DIR, recursive=True)
+        logging.info('Successfully deleted directories recursively.')
+
+    if create_dir:
+        logging.info('Creating directory to store dataframes...')
+        utils.create_dir(path.DATAFRAME_DIR)
+
+        logging.info('Creating directory to store models...')
+        utils.create_dir(path.MODEL_DIR)
+
+        logging.info('Creating directory to store candidate sets...')
+        utils.create_dir(path.CANDIDATE_SET_DIR)
+
+        logging.info('Creating directory to store RDD checkpoints...')
+        utils.create_dir(path.CHECKPOINT_DIR)
+
+        print('Done!')
+
+@cli.command(name='dataframe')
```

Comment:
Each of these 5 invocations below all do exception catching on their own, yes?

Reply:
Yes